src/main.rs

```diff
@@ -121,23 +121,23 @@ fn handle_move(prev_level: &types::Level, direction: types::Direction) -> types:
         }
     }
 
-    let mut chest_move = None;
+    let mut ball_move = None;
     if let Some((_, player_coords)) = player_move.clone() {
         for (index, entity) in level.entities.iter().enumerate() {
-            if let types::Entity::Chest(chest) = entity {
-                // if there is a chest where the player wants to move see if we can
+            if let types::Entity::Ball(ball) = entity {
```

Comment:
Lets call it SokoBox instead.

Reply:
done. changed PlayerGoal to PlayerAndGoal in tokens